classes/class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -69,15 +69,34 @@ private function store_origin_address( $settings ) {
 
 	public function update_items( $request ) {
 		$request_body = $request->get_body();
-		$settings = json_decode( $request_body, false, WOOCOMMERCE_CONNECT_MAX_JSON_DECODE_DEPTH );
+		$settings = json_decode( $request_body, TRUE, WOOCOMMERCE_CONNECT_MAX_JSON_DECODE_DEPTH );
 
 		$response = $this->api_client->send_shipping_label_request( $settings );
 
 		if ( true /* TODO: Trigger this only if the origin address correctly validates */ ) {
 			$this->store_origin_address( $settings );
 		}
 
-		return $response;
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$labels_data = array();
+		foreach ( $response->labels as $index => $label_data ) {
```

Comment:
You can't, since at this point I couldn't make a "valid" request to the `purchase-label` endpoint. I left this code here as a kind-of placeholder.